### PR TITLE
Simple hydro

### PR DESCRIPTION
**Pull request by zaphod-bx**
SimpleHydro without differences in tidal inundation duration created

### CODE REVIEW

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_diffusion_frac.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+
+fraction of water exchanged with neighbouring 4 gridcells per time step[-].
```
Comment (jbathmann) [inline Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_diffusion_frac.md:3]: volume or mass or percent? equally distributed among the 4 cells?

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_dilution_frac_upper.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
+
+fraction of water exchanged with seawater per time step at highest elevation[-].
```
Comment (jbathmann) [inline Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_dilution_frac_upper.md:3]: same question as for s_dilution_frac_lower

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_ini_sal.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
+
+initial salinity in the soil in ppt.
```
Comment (jbathmann) [inline Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_ini_sal.md:3]: which units?

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_slope.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+
+
+slope in y-direction.
```
Comment (jbathmann) [inline Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_slope.md:3]: is it possible to give a function z(y) or sth like slope(y), could you implement this?

File: test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        <type> NONE </type>
+    </visualization>
+    <tree_output>
+        <type> NONE </type>
+        <output_each_nth_timestep> 1 </output_each_nth_timestep>
+        <output_dir> ./ProjectLib/ExampleSetups/testoutputs/ </output_dir>
+        <geometry_output> r_stem </geometry_output>
+        <geometry_output> h_stem </geometry_output>
+        <geometry_output> r_crown </geometry_output>
+        <geometry_output> r_root </geometry_output>
```
Comment (jbathmann) [inline test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml:87]: you can delete those entries

File: test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    <tree_dynamics>
+        <aboveground_competition>
+            <type> SimpleTest </type>
+            <domain>
+                <x_1> 0 </x_1>
+                <y_1> 0 </y_1>
+                <x_2> 5 </x_2>
+                <y_2> 5 </y_2>
+	        <x_resolution> 20 </x_resolution>
+	        <y_resolution> 20 </y_resolution>
```
Comment (jbathmann) [inline test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml:12]: for the simpletest case, we don't need those entries

File: TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+        self.salinity += self._sea_salinity * water_loss / self.volume
+        self.salinity += (-self.salinity * self._dilution_frac_upper +
+                          self._sea_salinity * self._dilution_frac_upper)
+
+        # diffusion
+        salinity_new = self.salinity * (1 - self._diffusion_frac)
+        # x-dir
+        for ii in range(self.x_resolution):
+            if ii == self.x_resolution - 1:
+                salinity_new[:,ii] += (
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py:64]: reflecting bc?

File: TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+                salinity_new[ii + 1, :] += (
+                         self.salinity[ii, :] * self._diffusion_frac / 4)
+                salinity_new[ii, :] += (
+                         self.salinity[ii, :] * self._diffusion_frac / 4)
+            else:
+                salinity_new[ii + 1, :] += (
+                         self.salinity[ii, :] * self._diffusion_frac / 4)
+                salinity_new[ii - 1, :] += (
+                         self.salinity[ii, :] * self._diffusion_frac / 4)
+        self.salinity = salinity_new
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py:96]: nice.

File: TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+        # gradient-flow
+        multi_fac = self.Q_fac * (self._t_end - self._t_ini)
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py:98]: why a capital "Q"? please change according to our format style -> q_fac

**Review by jbathmann — COMMENTED** [review]
Can you please update the TML/BC/__init__.py?

Looks cool!
Do you need help for the implementation of the dilution factor(y)?

File: test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml [hunk 10]
```diff
@@ -18,20 +18,21 @@
...
+            <dilution_frac_lower> 0.2 </dilution_frac_lower>
+            <diffusion_frac> 0.05 </diffusion_frac>
             <sea_salinity> 35 </sea_salinity>
-            <ini_sal> 35 </ini_sal>
+            <ini_sal> 45 </ini_sal>
             <up_sal> 10 </up_sal>
-            <slope> 0.002 </slope>
+            <slope> 0.005 </slope>
             <k_f> 0.00025 </k_f>
+            <flooding_duration> 7.0, 4.1, 2.0, 1.2, 0.8, 0.6, 0.4, 0.2, 0.0  </flooding_duration>
```
Comment (jbathmann) [inline test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml:35]: wie werden die stützstellen aufgeteilt? das verstehe ich noch nicht aus deine documentation

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_diffusion_frac.md (hunk 1 above)
Comment (zaphod-bx) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_diffusion_frac.md:3]: wasser mit salz drin, hab egal ob 1/4der Masse oder des Volumens. den rest habsch erklärt.

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_dilution_frac_upper.md (hunk 2 above)
Comment (zaphod-bx) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_dilution_frac_upper.md:3]: dito

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_ini_sal.md (hunk 3 above)
Comment (zaphod-bx) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_ini_sal.md:3]: ppt

File: Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_slope.md (hunk 4 above)
Comment (zaphod-bx) [reply Documentation/ProjectFile/MangaProject/tree_dynamics/belowground_competition/SimpleHydro/s_slope.md:3]: good point! I could do it alike the dilution ...
Do you have it in the OGS?

File: test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml (hunk 6 above)
Comment (zaphod-bx) [reply test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml:12]: done

File: test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml (hunk 5 above)
Comment (zaphod-bx) [reply test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml:88]: done

File: TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py (hunk 7 above)
Comment (zaphod-bx) [reply TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py:66]: yes!

File: TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py (hunk 8 above)
Comment (zaphod-bx) [reply TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py:98]: wicked!

File: TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py (hunk 9 above)
Comment (zaphod-bx) [reply TreeModelLib/BelowgroundCompetition/SimpleHydro/SimpleHydro.py]: oköh

File: test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml (hunk 10 above)
Comment (zaphod-bx) [reply test/SmallTests/Test_Setups_small/SimpleHydro_SAZOI_BETTINA.xml:35]: interpol. gleiche abstände!

**Comment by zaphod-bx** [comment]
new submit!

**Pull request merged**